camunda-kafka-polling-client/src/main/java/de/viadee/camunda/kafka/pollingclient/service/polling/jdbc/CamundaJdbcPollingServiceImpl.java

```diff
@@ -236,6 +237,71 @@ public CamundaJdbcPollingServiceImpl(HistoryService historyService, RepositorySe
                              .map(historicIdentityLinkLog -> createIdentityLinkEventFromDetails(historicIdentityLinkLog))::iterator;
     }
 
+    @Override
+    public Iterable<DecisionDefinitionEvent> pollDecisionDefinitions(Date deploymentAfter, Date deploymentBefore) {
+        // TODO check implementation
+
+        // There seems to be a slight bug in Camunda SQL queries regarding deployments.
+        // Where the other history queries regarding time boundaries are inclusive (startedBefore, startedAfter, ...),
+        // deploymentBefore and deploymentAfter are implemented exclusive.
+        // Thus we have to slightly adjust the deploymentAfter parameter by 1 millisecond to act inclusive:
+        deploymentAfter = new Date(deploymentAfter.getTime() - 1);
+
+        // query deployments
+        List<Deployment> deployments = repositoryService.createDeploymentQuery()
+                                                        .deploymentAfter(deploymentAfter)
+                                                        .deploymentBefore(deploymentBefore)
+                                                        .list();
+
+        List<DecisionDefinitionEvent> result = new ArrayList<>();
+
+        for (Deployment deployment : deployments) {
+            List<DecisionDefinition> decisionDefinitions = repositoryService.createDecisionDefinitionQuery()
+                                                                          .deploymentId(deployment.getId())
+                                                                          .list();
+
+            // query proc def
+            for (DecisionDefinition decisionDefinition : decisionDefinitions) {
+                DecisionDefinitionEvent decisionDefinitionEvent = createDecisionDefinitionEvent(deployment,
+                                                                                             decisionDefinition);
+
+                // query xml
+                try {
+                    String xml = IOUtils.toString(repositoryService.getResourceAsStream(decisionDefinition.getDeploymentId(),
+                                                                                        decisionDefinition.getResourceName()));
+                    decisionDefinitionEvent.setXml(xml);
+                } catch (IOException e) {
+                    throw new RuntimeException(
+                                               "error while reading xml for decision definition "
+                                                       + decisionDefinition.getId(),
+                                               e);
+                }
+
+                result.add(decisionDefinitionEvent);
+            }
+
+        }
+
+        return result;
+    }
+    
+
+    @Override
+    public Iterable<DecisionInstanceEvent> pollDecisionInstances(ActivityInstanceEvent activityInstanceEvent) {
+        // TODO implement
+        return null;
```

Comment:
This might produce NullPointerExceptions. Maybe we could just raise an UnsupportedOperationException?

Reply:
I would rather say we should implement this (and the other two methods for polling Inputs and Outputs) before we get this merged shouldn't we? 

Otherwise the Polling-Client would not be useable with JDBC-Configuration as the method pollDecisionInstances is always called. Cf. [here ](https://github.com/viadee/camunda-kafka-polling-client/blob/36500bcd721b38f08bb10eaba620c62735d0a55e/camunda-kafka-polling-client/src/main/java/de/viadee/camunda/kafka/pollingclient/job/runtime/RuntimeDataPollingService.java#L241) and [here](https://github.com/viadee/camunda-kafka-polling-client/blob/36500bcd721b38f08bb10eaba620c62735d0a55e/camunda-kafka-polling-client/src/main/java/de/viadee/camunda/kafka/pollingclient/job/runtime/RuntimeDataPollingService.java#L160)

Reply:
I totally agree. This would be the best option.